packages/backend/src/nest/connections-manager/connections-manager.service.ts

```diff
@@ -260,144 +311,205 @@ export class ConnectionsManagerService extends EventEmitter implements OnModuleI
       return
     }
 
+    // TODO: Should we save this network info in LevelDB at this point?
     return network
   }
 
-  private async generatePSK() {
-    const pskBase64 = Libp2pService.generateLibp2pPSK().psk
-    await this.localDbService.put(LocalDBKeys.PSK, pskBase64)
-    this.logger('Generated Libp2p PSK')
-    this.serverIoProvider.io.emit(SocketActionTypes.LIBP2P_PSK_STORED, { psk: pskBase64 })
-  }
-
-  public async createCommunity(payload: InitCommunityPayload) {
+  public async createCommunity(payload: InitCommunityPayload): Promise<Community | undefined> {
     this.logger('Creating community: peers:', payload.peers)
 
-    await this.generatePSK()
+    if (!payload.CA || !payload.rootCa) {
+      this.logger.error('CA and rootCa are required to create community')
+      return
+    }
+
+    if (!payload.ownerCsr) {
+      this.logger.error('ownerCsr is required to create community')
+      return
+    }
+
+    const psk = Libp2pService.generateLibp2pPSK().psk
+    let ownerCertResult: SavedOwnerCertificatePayload
+
+    try {
+      this.emit(SocketActionTypes.CONNECTION_PROCESS_INFO, ConnectionProcessInfo.REGISTERING_OWNER_CERTIFICATE)
+      ownerCertResult = await this.registrationService.registerOwnerCertificate({
+        communityId: payload.id,
+        userCsr: payload.ownerCsr,
+        permsData: {
+          certificate: payload.CA.rootCertString,
+          privKey: payload.CA.rootKeyString,
+        },
+      })
+    } catch (e) {
+      this.logger.error('Failed to register owner certificate')
+      return
+    }
+
+    const localAddress = createLibp2pAddress(payload.hiddenService.onionAddress, payload.peerId.id)
 
-    await this.launchCommunity(payload)
-    this.logger(`Created and launched community ${payload.id}`)
-    this.serverIoProvider.io.emit(SocketActionTypes.COMMUNITY_CREATED, { id: payload.id })
+    const community = {
+      id: payload.id,
+      name: payload.name,
+      CA: payload.CA,
+      rootCa: payload.rootCa,
+      peerList: [localAddress],
+      ownerCertificate: ownerCertResult.network.certificate,
+      psk: psk,
+    }
+
+    const network = {
+      hiddenService: payload.hiddenService,
+      peerId: payload.peerId,
+    }
+
+    await this.localDbService.setCommunity(community)
+    await this.localDbService.setCurrentCommunityId(community.id)
+    // TODO: Revisit this when we move the Identity model to the backend, since
+    // this network data lives in that model.
+    await this.localDbService.setNetworkInfo(network)
+
+    await this.launchCommunity({ community, network })
+    this.logger(`Created and launched community ${community.id}`)
+
+    return community
   }
 
-  public async launchCommunity(payload: InitCommunityPayload) {
-    this.logger('Launching community: peers:', payload.peers)
-    this.communityState = ServiceState.LAUNCHING
+  public async joinCommunity(payload: InitCommunityPayload): Promise<Community | undefined> {
```

Comment:
joinCommunity is a new function so that we can focus on validation and creating a Community object, while launchCommunity is still used but it's only for launching an already-validated community.